packages/scope-manager/src/referencer/VisitorBase.ts

```diff
@@ -67,6 +67,7 @@ abstract class VisitorBase {
    * Dispatching node.
    */
   visit(node: TSESTree.Node | null | undefined): void {
+    // eslint-disable-next-line @typescript-eslint/strict-enums
     if (node == null || node.type == null) {
```

Comment:
Some of this code came from the `eslint-scope` package from which the code was forked.

I think this is just a guard against the argument to `visit` possibly being a non-node?
I.e. in case an array got passed in or something.

I don't think that case is ever hit in our codebase or else our types would complain.

---

Side-note - should the rule be complaining about this case?
It seems weird for the rule to be complaining about comparisons with `null`/`undefined`?

Reply:
A few days ago, I whitelisted comparisons to null/undefined, so something must have gone wrong.
At any rate, it seems to be fixed now.